extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Converters;
+using System.Collections.Generic;
+using Microsoft.Azure.Cosmos;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind Cosmos DB type parameters.
+    /// </summary>
+    internal class CosmosDBConverter : IInputConverter
+    {
+        private readonly IOptionsSnapshot<CosmosDBBindingOptions> _cosmosOptions;
+
+        public CosmosDBConverter(IOptionsSnapshot<CosmosDBBindingOptions> cosmosOptions)
+        {
+            _cosmosOptions = cosmosOptions ?? throw new ArgumentNullException(nameof(cosmosOptions));
+        }
+
+        public async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is ModelBindingData modelBindingData)
+            {
+                if (modelBindingData.Source is not Constants.ExtensionName)
+                {
+                    return ConversionResult.Unhandled();
+                }
+
+                try
+                {
+                    var cosmosAttribute = modelBindingData.Content.ToObjectFromJson<CosmosDBInputAttribute>();
+                    object result = await ToTargetType(context.TargetType, cosmosAttribute);
+
+                    if (result is not null)
+                    {
+                        return ConversionResult.Success(result);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // What do we want to do for error handling?
```

Comment:
log using ILogger ? You can DI `ILogger<CosmosDBConverter>` into the constructor of CosmosDBConverter